TunnelKit/Sources/Protocols/OpenVPN/TLSBox.m

```diff
@@ -216,31 +245,47 @@ - (BOOL)startWithError:(NSError *__autoreleasing *)error
     if (self.securityLevel != TLSBoxDefaultSecurityLevel) {
         SSL_CTX_set_security_level(self.ctx, (int)self.securityLevel);
     }
-    if (!SSL_CTX_load_verify_locations(self.ctx, [self.caPath cStringUsingEncoding:NSASCIIStringEncoding], NULL)) {
-        ERR_print_errors_fp(stdout);
-        if (error) {
-            *error = TunnelKitErrorWithCode(TunnelKitErrorCodeTLSCertificateAuthority);
+
+    if (self.caPEM) {
+        BIO *bio = create_BIO_from_PEM(self.caPEM);
+        X509 *ca = PEM_read_bio_X509(bio, NULL, NULL, NULL);
+        BIO_free(bio);
+        X509_STORE *trustedStore = SSL_CTX_get_cert_store(self.ctx);
+        if (!X509_STORE_add_cert(trustedStore, ca)) {
+            ERR_print_errors_fp(stdout);
+            if (error) {
+                *error = TunnelKitErrorWithCode(TunnelKitErrorCodeTLSCertificateAuthority);
+            }
+            return NO;
```

Comment:
According to L274, `ca` should also be deallocated.

Reply:
Agreed. Good catch.